src/50_Samples_%26_Templates/Product.html

```diff
@@ -313,11 +415,20 @@ <h5 class="price-description">Price: $1.99</h5>
         </div>
         <div>
           <label for="size">Size:</label>
-          <amp-selector name="size" layout="container">
+          <amp-selector name="size" layout="container" on="select:AMP.setState(selectedSize=event.targetOption)" [selected]="allColors[selectedCarousel].availableSizes.indexOf(selectedSize) >= 0 ? selectedSize : allColors[selectedCarousel].availableSizes[0]">
             <ul>
-              <li option="S">S</li>
-              <li option="M" selected>M</li>
+              <div class="size-container">
+                <li option="S">S</li>
+                <span class="show-size-unavailable" [class]="allColors[selectedCarousel].availableSizes.indexOf('S') >= 0 ? 'hide' : 'show-size-unavailable'">X</span>
```

Comment:
Can we use CSS to show when a size is unavailable? This makes it also possible to format the X to better fit into the box. Another option is to simply remove the shadow and lower the opacity.